server.js

```diff
@@ -7965,6 +7965,38 @@ cache(function (data, match, sendBadge, request) {
   });
 }));
 
+// Dependabot SemVer compatibility integration
+camp.route(/^\/dependabot\/semver\/([^/]+)\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  const packageManager = match[1];
+  const dependencyName = match[2];
+  const format = match[3];
+  const options = {
+    method: 'GET',
+    headers: { 'Accept': 'application/json' },
+    uri: `https://api.dependabot.com/badges/compatibility_score?package-manager=${packageManager}&dependency-name=${dependencyName}&version-scheme=semver`
+  };
+  const badgeData = getBadgeData('semver compatibility', data);
+  badgeData.links = [`https://dependabot.com/compatibility-score.html?package-manager=${packageManager}&dependency-name=${dependencyName}&version-scheme=semver`];
+  badgeData.logo = getLogo('dependabot', data);
+  request(options, function(err, res) {
+    if (checkErrorResponse(badgeData, err, res)) {
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      const dependabotData = JSON.parse(res['body']);
+      badgeData.text[1] = dependabotData.status;
+      badgeData.colorscheme = dependabotData.colour;
+      sendBadge(format, badgeData);
+    } catch(e) {
+      badgeData.text[1] = 'invalid';
```

Comment:
@greysteil This fragment of code does not have tests. Would you like to add one? You can use mocked response with invalid JSON. An example: https://github.com/badges/shields/blob/ea4b758612c6e87ba227773fddee7ceab03ee138/services/bountysource/bountysource.tester.js#L30-L36